src/cpp/participant/Participant.cpp

```diff
@@ -18,34 +18,44 @@
 namespace eprosima {
 namespace uxr {
 
-Participant::Participant(const dds::xrce::ObjectId& id, Middleware* middleware) : XRCEObject(id, middleware) {}
-
-Participant::~Participant()
+Participant* Participant::create(
+        const dds::xrce::ObjectId& object_id,
+        const dds::xrce::OBJK_PARTICIPANT_Representation& representation,
+        Middleware& middleware)
 {
-    middleware_->delete_participant(get_raw_id());
-}
+    bool created_entity = false;
+    uint16_t raw_object_id = uint16_t((object_id[0] << 8) + object_id[1]);
```

Comment:
Like the previous comment, maybe exists a way to factorize this code into a static method of `XRCEObject` or something. Only the `XRCEObject` will know how to perform these conversions.

Reply:
Maybe this function could be called from the `ObjectIDHash::operator()` function too.

Reply:
Done! Thank you for the advice and help!